adds parser to .js files

RafaelaMG opened the pull request:
When using skeleton-parser now it will be possible to define something like this:

```javascript
const metaData = await skeletonParser({
    cwd: ".",
    folders: ["components", "layouts"],
    yml: false //Parses .js files instead of .yml
  });
```
This enables parsing of .js files.

The default yml definition in skeleton-parser is set to true and unless it is specified as false it will only parse .yml files.

RafaelaMG committed:
adds parser to .js files

lucalanca commented:
@RafaelaMG can you add tests for this change? You can find examples in the test folder https://github.com/lucalanca/skeleton-parser/blob/master/test/index.js 

You'll see that this PR is failing the tests.

lucalanca commented:
another thing, I see that this PR is quite big due to style changes. I think going foward, we can use prettier options, but I would introduce this style changes on a separate PR to make it easier to read this one.

RafaelaMG committed:
wip

RafaelaMG committed:
fixs and adds tests

RafaelaMG committed:
fixes and adds more tests

RafaelaMG committed:
updates node version

RafaelaMG committed:
fix CI node version

RafaelaMG commented:
I fixed the tests that were failing and I also added a couple more to test the parser with definition.js files.
I also used the xo linter to fix the styling and formatting so the comparison between code changes should be "readable" now.

RafaelaMG committed:
adds src default option

RafaelaMG committed:
adds script for ES2005 conversion

RafaelaMG committed:
fix babel plugins

RafaelaMG committed:
fix for CI tests

RafaelaMG committed:
updates readme

RafaelaMG committed:
updates readme

lucalanca commented:
@RafaelaMG can you just remove `package-lock.json`? I would add it (though I would prefer `yarn.lock` but not on this PR 😸 )

RafaelaMG committed:
removes package-lock.json

RafaelaMG commented:
Done 👍
